### ISSUE TITLE
Slack configuration

### ISSUE THREAD

**Issue by Jonher937** [rycus86/release-watcher#16]
Hi, we currently use this with a basic webhook in Slack, however icon and username are not allowed to be overridden based on the [Slack documentation](https://api.slack.com/messaging/webhooks#advanced_message_formatting)
> You **cannot** override the default channel (chosen by the user who installed your app), username, or icon when you're using Incoming Webhooks to post messages. Instead, these values will always inherit from the associated Slack app configuration.

Would you be open for a PR to replace Webhook with [Slack Chat Post Message API](https://slack.com/api/chat.postMessage)?
This would allow to override the icon, username, channel, etc.. 

Or have you found a way to achieve this with the current code that I'm missing? (I noticed you mentioned you were using a custom icon in a previous PR)

**Comment by rycus86** [comment]
Hm, that's strange, the custom icon seems to work OK for me.
![Screenshot_20210708-212613](https://user-images.githubusercontent.com/3105242/124914120-53464480-e033-11eb-98bd-6e507d60c903.jpg)

Do you get an error if you try to set them?

**Comment by Jonher937** [comment]
So I have created an app in Slack and enabled Incoming webhook:
![2021-07-08-133542_scrot](https://user-images.githubusercontent.com/1125285/124915245-9c75a480-dff1-11eb-9ded-ee6719bf6efd.png)

My URL looks like: `https://hooks.slack.com/services/<some-id>/<some-id>/<some-id>`

So if I post using curl to test:
```bash
curl -X POST -H 'Content-type: application/json' --data '{"text":"Hello, World!", "icon_url": "https://upload.wikimedia.org/wikipedia/commons/thumb/b/bc/Refresh_icon.png/240px-Refresh_icon.png"}' https://hooks.slack.com/services/<some-id>/<some-id>/<some-id>
```
then I get only the default app-icon back from Slack, just like the documentation suggests.
![2021-07-08-134333_scrot](https://user-images.githubusercontent.com/1125285/124916061-887e7280-dff2-11eb-824b-91a77c74a91b.png)

Maybe the difference is that you have one Slack App per Webhook rather than one App that can override the icon based on the json payload?

**Comment by rycus86** [comment]
I checked your curl against my integration, and that set the icon URL fine. I noticed though that mine is set up as `Incoming WebHooks` integration and it now has a big warning, saying it's a "legacy custom integration" so it looks like Slack might have something newer nowadays.

Try to see if you can still find it under `Manage` > `Custom Integrations` > `Incoming WebHooks`.
Alternatively, I'm happy to review PRs adding support for the new kind of webhook integration if that's more desirable.

**Comment by Jonher937** [comment]
That would be the difference indeed, I did not use the legacy "Incoming WebHooks", instead I used a Slack App like recommended from Slack: "Instead, we suggest that you check out their replacement: Slack apps".
At least this makes sense why it works for you and not me :slightly_smiling_face: 

I'll close this issue for now and see if I spend some time changing the Slack notifier to support sending to the Message API instead. The main difference would be the URL to post to and including a Bearer token in the header of the request.
Thanks!
